awsapilib/authentication/authentication.py

```diff
@@ -509,33 +509,40 @@ def get_billing_authenticated_session(self):
 
         """
         service = 'billing'
-        url = f'{self.urls.console}/{service}/home?region={self.region}'
         self._get_response(self.get_signed_url())
-        host = urllib.parse.urlparse(url)[1]
-        self.logger.debug('Setting host to: %s', host)
-        self._get_response(url, extra_cookies=[FilterCookie('JSESSIONID', f'/{service}'),
-                                               FilterCookie('aws-userInfo-signed', )])
-        hash_args = self._get_response(url,
-                                       params={'state': 'hashArgs#'},
-                                       extra_cookies=[FilterCookie('JSESSIONID', f'/{service}'),
-                                                      FilterCookie('aws-userInfo-signed', ),
-                                                      FilterCookie('aws-creds-code-verifier', f'/{service}')])
-        oauth = self._get_response(hash_args.headers.get('Location'),
+
+        regional_url = f'{self.urls.scheme}us-east-1.console.{self.urls.root_domain}/billing/home'
```

Comment:
Wouldn't it be nicer to extend the Urls class to be able to provide the regional root console url so the region is not hard coded in this request? Maybe even the regional billing url, since the billing console is provided?